pkg/volume/iscsi/iscsi.go

```diff
@@ -98,58 +95,89 @@ func (plugin *iscsiPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 func (plugin *iscsiPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
 	// Inject real implementations here, test through the internal function.
 	var secret map[string]string
-	source, _, err := getVolumeSource(spec)
+	if pod == nil {
+		return nil, fmt.Errorf("nil pod")
+	}
+	chapDiscover, chapSession, err := getISCSICHAPInfo(spec)
```

Comment:
same here, why the newly added check for chapDiscover/chapSession? is a secret required in all cases if those are set? is a secret ignored in all cases if those are not set?

Reply:
see above. Secret should be ignored if CHAP is not enabled.